### DISCUSSION TITLE
.security subfield problem with wazuh logs

### DISCUSSION THREAD

**Discussion by YannickZ5** [Security-Onion-Solutions/securityonion#7626] [Unsupported Versions] [answered]
Hallo everyone!

First of all, here are some informations about our deployment.

- SO-Version: 2.3.110
- Standalone
- Airgap
- Install from ISO
- No errors when I run so-status

Recently I switched from winlogbeat to wazuh on our windows clients and servers. Now I am facing a problem with some of our self created sigma rules. I have created an example that hopefully explains the problem:

If I create a sigma rule with the following arguments for example:

```
...
logsource:
    category: process_creation
    product: windows
detection:
    selection:
        Image: "C:\\Windows\\System32\\svchost.exe"
    condition: selection
...
```

The created elastic query looks like this:

`(event.code.security:"1" AND winlog.channel.security:"Microsoft\-Windows\-Sysmon\/Operational" AND process.executable.security:"C\:\\Windows\\System32\\svchost.exe")`

When I run this query in Kibana with lucene, Kibana can not provide me the corresponding log entrys from wazuh. Only the older winlogbeat entrys come up. But when I delete the `.security` field from the `process.executable` query, Kibana then shows me the wazuh logs and the winlogbeat logs.

The values in Elasitcsearch for the process.executable entry are different but I think this should not be a problem:

Wazuh:
`proccess.executable: C:\\Windows\\System32\\svchost.exe`
Winlogbeat:
`proccess.executable: C:\Windows\System32\svchost.exe`

What I also noticed is that querys without a full path still works. For example: `Image: "*\whoami.exe"` is still working with the `.security` subfield. 

We are using the full path information for file auditing and some other stuff.

Thank you for your help and support!

**Comment by defensivedepth** [comment, marked as answer]
A couple things to note:

- Typically you do not need the double backslashes in the Sigma rule 

- Use single quotes for Sigma rules; Sigma rules use YAML, so `"` will work, but can introduce some unexpected issues when special sequences/chars are used

- There is currently a bug in Wazuh that is adding extra backlashes - there is nothing we can do about this until they fix it upstream: https://github.com/wazuh/wazuh/issues/4509
